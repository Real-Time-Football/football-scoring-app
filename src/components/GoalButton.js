import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const DefaultButton = styled.button`
    background-color: #4caf50;
    border: none;
    color: white;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
`

const GoalButton = ({ Button, ...props }) => <Button {...props} />

GoalButton.propTypes = {
    children: PropTypes.node.isRequired,
    Button: PropTypes.elementType,
}

GoalButton.defaultProps = {
    Button: DefaultButton,
}

export default GoalButton
