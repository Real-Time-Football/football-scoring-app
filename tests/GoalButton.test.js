import React from 'react'
import Enzyme, { mount } from 'enzyme'
import GoalButton from '../src/components/GoalButton'

describe('Goal Button', () => {
    const text = 'Button text'
    let wrapper

    beforeEach(() => {
        wrapper = Enzyme.shallow(<GoalButton>{text}</GoalButton>)
    })

    it('renders a <button>', () => {
        const mounted = Enzyme.mount(<GoalButton data-team-type={'home'}>{text}</GoalButton>)
        expect(mounted.containsMatchingElement(<button data-team-type={'home'}>{text}</button>)).toBe(true)
    })

    it('passes `children` to through the <button>', () => {
        expect(wrapper.prop('children')).toBe(text)
    })

    it('passes other props through to the <button>', () => {
        const onClick = () => {}
        const className = 'goal-button'
        const dataTeamType = 'home'
        wrapper.setProps({ onClick, className, 'data-team-type': dataTeamType })
        expect(wrapper.prop('onClick')).toBe(onClick)
        expect(wrapper.prop('className')).toBe(className)
        expect(wrapper.prop('data-team-type')).toBe(dataTeamType)
    })
})
