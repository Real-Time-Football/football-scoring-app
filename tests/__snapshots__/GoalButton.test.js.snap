// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  background-color: #c70039;
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 2px;
}

<GoalButton__DefaultButton
  data-team-type="home"
>
  <button
    className="c0"
    data-team-type="home"
  />
</GoalButton__DefaultButton>
`;

exports[`Goal Button renders correctly 1`] = `
<GoalButton__DefaultButton
  data-team-type="home"
>
  Button text
</GoalButton__DefaultButton>
`;
